designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,287 @@
+
+## Motivation
+
+Kubebuilder users currently face the challenges of lacking an officially supported distribution mechanism for Helm Chart. They seek to:
+
+- Harness the power of Helm Chart as a package manager for the Project, enabling seamless adaptation to diverse deployment environments.
+- Take advantage of Helm's dependency management capabilities to simplify the installation process of Project dependencies, such as cert-manager.
+- Seamlessly integrate with Helm's ecosystem, including FluxCD, to efficiently manage the Project.
+
+However, Kubebuilder does not offer any utilities to facilitate the distribution of projects currently. One could clone the entire project and utilize the Makefile targets to apply the solution on clusters. Consequently, this proposal aims to introduce a method that allows Kubebuilder users to easily distribute their projects through Helm Charts, a strategy that many similar many well-known projects contains the distribution of helm charts:
+
+- [mongodb](https://artifacthub.io/packages/helm/mongodb-helm-charts/community-operator
+)
+- [cert-manager](https://cert-manager.io/v1.6-docs/installation/helm/#1-add-the-jetstack-helm-repository)
+- [prometheus](https://bitnami.com/stack/prometheus-operator/helm)
+- [aws-load-balancer-controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller/tree/main/helm/aws-load-balancer-controller)
+
+
+**NOTE** For further context see the [discussion topic](https://github.com/kubernetes-sigs/kubebuilder/discussions/3074)
+
+### Goals
+
+- Allow Kubebuilder users distribute their projects using Helm easily.
+- Make the best effort to preserve any customizations made to the Helm Charts by the users, which means we will skip syncs in the `values.ymal`.
+- Stick with Helm layout definitions and externalize into the relevant values-only options to distribute the default scaffold done by Kubebuilder. We should follow https://helm.sh/docs/chart_best_practices.
+
+### Non-Goals
+
+- Converting any Kustomize configuration to Helm Charts like [helmify](https://github.com/arttor/helmify) does.
+- Support the deprecated plugins. This option should be supported from go/v4 and kustomize/v2
+- Introduce support for Helm in addition to Kustomize, or replace Kustomize with Helm entirely, similar to the approach taken by Operator-SDK, thereby allowing users to utilize Helm Charts to build their Project.
+- Attend standard practices that deviate from Helm Chart layout, definition, or conventions to workaround its limitations.
+- Provide values and options to allow users to perform customizations on the Helm Chart of features that are not used by default in the Project layout.
```

Comment:
```suggestion
```

I think we can remove it. Above we are saying that we will try to update the files. So, it does not seems to make sense